src/core/api_client/tests/types.ts

```diff
@@ -0,0 +1,18 @@
+import type { ApiMethod, GeneralApiProblem } from '../types';
+
+/**
+ * Standard API error response shape for API client tests
+ */
+export type ApiErrorResponse = GeneralApiProblem & {
```

Comment:
NIT: Maybe it's better to add prefix to these types to clearly distinct they're for test mocks? (e.g. MockApiErrorResponse etc)